### ISSUE TITLE
Using Admin resetOffsets to earliest makes consumerGroup offset reset to latest

### ISSUE THREAD

**Issue by joaosouzaminu** [tulios/kafkajs#1020]
**Describe the bug**
Calling `admin.resetOffsets({ groupId, topic, earliest: true })` then starting a consumer for that groupId makes it consume all messages from beginning.

**To Reproduce**

[Link to the example project reproducing the issue](https://github.com/joaosouzaminu/reset-topic-offset)

**Expected behavior**
The consumer should not consume any messages after the `seek` made by the admin client.

> Example: 
> Say Producer wrote 3 messages to a topic with the Consumer stoped and Admin resetOffsets was executed with `earliest: true`, then the Consumer should not recieve any messages.

**Observed behavior**
After starting the Consumer it gets all messages from the topic.

**Environment:**
 - OS: Ubuntu 20.04 running Kafka with Confluent Docker Image v6.0.1
 - KafkaJS version: 1.15.0
 - NodeJS version: 14.15.1

**Comment by Nevon** [comment]
I don't really understand the issue you are describing. To me, this sounds like it's working exactly as intended.

1. You have a partition with 1 message on it
2. A consumer consumes that message and commits the offset for it.
3. You reset the consumer group's offset back to `earliest`.
4. Now the consumer consumes from the earliest offset again.

Is there something I'm missing here?

**Comment by tulios** [comment]
You probably want `earliest: false`, so it moves to the last offset of the topic instead of the first.

**Comment by joaosouzaminu** [comment]
Thanks @Nevon and @tulios for the quick answer.

I think I might have got wrong the meaning of the parameter. But anyways, I've tested with `earliest: false` and got the exact same behaviour as with `earliest: true`.

Actually that was the first attempt I did and thought I might be doind something wrong. I'll update the issue description and the example code and accept further suggestions.

**Comment by joaosouzaminu** [comment]
Just adding a bit of info from the tests me and my team have been doing today.

It seems that this flow of `producing > reseting offset > starting consumer` behaves as we expected when the Consumer subscribes to the topic using `fromBeginning: false`.

Is this the expected behaviour? I mean, when the subscription uses `fromBeginning: true` the consumer should get all messages from the topic even if its group offset was set to latest?

**Comment by tulios** [comment]
> when the subscription uses `fromBeginning: true` the consumer should get all messages from the topic even if its group offset was set to latest?

No. This flag only defines the default behavior, which happens whenever an offset is not found for the `groupId`. Once the offset is defined, this flag is not used.
